### PR TITLE
DEPR: Removing previously deprecated flavor parameter from SQLiteData…

### PR DESCRIPTION
**Pull request by datapythonista**
…base (#6581)

- [X] related to #6581
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -314,6 +314,7 @@ Removal of prior version deprecations/changes
 - The ``Panel4D`` and ``PanelND`` classes have been removed (:issue:`13776`)
 - The ``Panel``class has dropped the ``to_long``and ``toLong`` methods (:issue:`19077`)
 - The options ``display.line_with`` and ``display.height`` are removed in favor of ``display.width`` and ``display.max_rows`` respectively (:issue:`4391`, :issue:`19107`)
+- The ``flavor`` parameter have been removed from ``SQLiteDatabase`` and :func:`to_sql` method
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: just say ``to_sql``, ``SQLiteDatabase`` is not public. add the issue number and this issue.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
hmm, these are not building on travis for some reason.

**Comment by datapythonista** [comment]
It's a great reason for you to take a break today, it's Sunday! ;)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**